pkg/cloudprovider/providers/photon/photon.go

```diff
@@ -0,0 +1,473 @@
+		return "", err
+	}
+
+	for _, vm := range vmList.Items {
+		task, err := Esxclient.VMs.GetNetworks(vm.ID)
+		if err != nil {
+			glog.V(2).Info("Photon Cloud Provider: GetNetworks failed for vm.ID %s")
+		} else {
+			task, err = Esxclient.Tasks.Wait(task.ID)
+			if err != nil {
+				glog.V(2).Info("Photon Cloud Provider: , Wait task for GetNetworks failed for vm.ID %s")
+			} else {
+				networkConnections := task.ResourceProperties.(map[string]interface{})
+				networks := networkConnections["networkConnections"].([]interface{})
+				for _, nt := range networks {
+					network := nt.(map[string]interface{})
+					if val, ok := network["ipAddress"]; ok && val != nil {
+						ipAddr := val.(string)
+						if ipAddr == IPAddress {
+							return vm.ID, nil
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("Photon Cloud Provider: no matching VM is found with IP %s", IPAddress)
+}
+
+func getProjIDbyName(tenantName, projName string) (string, error) {
+	tenants, err := Esxclient.Tenants.GetAll()
+	if err != nil {
+		return "", err
+	}
+
+	for _, tenant := range tenants.Items {
+		if tenant.Name == tenantName {
+			projects, err := Esxclient.Tenants.GetProjects(tenant.ID, nil)
+			if err != nil {
+				return "", err
+			}
+
+			for _, project := range projects.Items {
+				if project.Name == projName {
+					return project.ID, nil
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("Photon Cloud Provider: no matching tenant/project name is found with %s/%s", tenantName, projName)
+}
+
+func newPCCloud(cfg PCConfig) (*PCCloud, error) {
+	if len(cfg.Global.CloudTarget) == 0 {
+		return nil, errors.New("Invalid Photon Controller endpoint")
+	}
+
+	//TODO: add handling of certification enabled situation
+	options := &photon.ClientOptions{
+		IgnoreCertificate: cfg.Global.IgnoreCertificate,
+	}
+
+	Esxclient = photon.NewClient(cfg.Global.CloudTarget, options, logger)
+	status, err := Esxclient.Status.Get()
+	if err != nil {
+		return nil, errors.New("Photon Controller new client creation failed")
+	}
+	glog.V(2).Info("Photon Cloud Provider: Status of the new photon controller client: %v", status)
+
+	// Get Photon Controller project ID for future use
+	projID, err := getProjIDbyName(cfg.Global.Tenant, cfg.Global.Project)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get local hostname for localInstanceID
+	cmd := exec.Command("bash", "-c", `echo $HOSTNAME`)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	if out.Len() == 0 {
+		return nil, fmt.Errorf("Photon Cloud Provider: unable to retrieve hostname for Instance ID")
+	}
+	hostname := strings.TrimRight(out.String(), "\n")
+	vmID, err := getVMIDbyNodename(projID, hostname)
+	if err != nil {
+		return nil, err
+	}
+
+	pc := PCCloud{
+		cfg:              &cfg,
+		localInstanceID:  vmID,
+		localHostname:    hostname,
+		localK8sHostname: "",
+		projID:           projID,
+	}
+
+	overrideIP = cfg.Global.OverrideIP
```

Comment:
Can you add comment on overrideIP either here or somewhere else? What this option is why is it needed? What is the default if user doesn't specify in config file?